internal/unionstore/art/art.go

```diff
@@ -47,20 +48,314 @@ func New() *ART {
 }
 
 func (t *ART) Get(key []byte) ([]byte, error) {
-	panic("unimplemented")
+	// 1. search the leaf node.
+	_, leaf := t.search(key)
+	if leaf == nil || leaf.vAddr.IsNull() {
+		return nil, tikverr.ErrNotExist
+	}
+	// 2. get the value from the vlog.
+	return t.allocator.vlogAllocator.GetValue(leaf.vAddr), nil
 }
 
 // GetFlags returns the latest flags associated with key.
 func (t *ART) GetFlags(key []byte) (kv.KeyFlags, error) {
-	panic("unimplemented")
+	_, leaf := t.search(key)
+	if leaf == nil {
+		return 0, tikverr.ErrNotExist
+	}
+	if leaf.vAddr.IsNull() && leaf.isDeleted() {
+		return 0, tikverr.ErrNotExist
+	}
+	return leaf.getKeyFlags(), nil
 }
 
-func (t *ART) Set(key artKey, value []byte, ops []kv.FlagsOp) error {
-	panic("unimplemented")
+func (t *ART) Set(key artKey, value []byte, ops ...kv.FlagsOp) error {
+	if value != nil {
+		if size := uint64(len(key) + len(value)); size > t.entrySizeLimit {
+			return &tikverr.ErrEntryTooLarge{
+				Limit: t.entrySizeLimit,
+				Size:  size,
+			}
+		}
+	}
+	if len(t.stages) == 0 {
+		t.dirty = true
+	}
+	// 1. create or search the exist leaf in the tree.
+	addr, leaf := t.recursiveInsert(key)
+	// 2. set the value and flags.
+	t.setValue(addr, leaf, value, ops)
+	if uint64(t.Size()) > t.bufferSizeLimit {
+		return &tikverr.ErrTxnTooLarge{Size: t.Size()}
+	}
+	return nil
 }
 
 func (t *ART) search(key artKey) (arena.MemdbArenaAddr, *artLeaf) {
-	panic("unimplemented")
+	current := t.root
+	if current == nullArtNode {
+		return arena.NullAddr, nil
+	}
+	depth := uint32(0)
+	var node *nodeBase
+	for {
+		if current.isLeaf() {
+			lf := current.leaf(&t.allocator)
+			if lf.match(0, key) {
```

Comment:
Yes, to avoid false positive optimistic match, we need `prefixLen := node.matchDeep(key, depth)` here (it's slower in my test).

https://github.com/tikv/client-go/blob/f0a768d65369136df3a78605368b549f5b6eeabd/internal/unionstore/art/art.go#L126